packages/core/src/amazonqFeatureDev/util/i18n-helper.ts

```diff
@@ -0,0 +1,17 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import translations from '../../../src/shared/i18n/package.nls.json'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+
+export const i18n = (code: string) => {
+    const translationsTyped = translations as { [key: string]: string }
+
+    if (translationsTyped[code]) {
+        return localize(code, translationsTyped[code])
```

Comment:
Why not just use localize(code,  "fallback")?

Reply:
Because we will keep the same strings in the code and its easy just fallback to the default english file without to duplicate the same tag in a fallback. At least makes more sense to me.

Reply:
Have you tried `vscode.l10n.t()` ?

Reply:
I agree that the new VS Code API have a more solid support using l10n. However, @chengoramazon the current Amazon Q implementation uses https://github.com/microsoft/vscode-nls, which is the same I'm reusing it. Use l10n now api would make we differ the current implementation and require we have another approach in all the other 175 files that its using nls.

Reply:
> [vscode-nls](https://github.com/microsoft/vscode-nls) ⚠️ This package is no longer receiving new features in favor of the new localization library, [vscode-l10n](https://github.com/microsoft/vscode-l10n). Please use that collection of libraries instead.

We don't need the fallback if we go with l10n, which would be proper.

Reply:
Obviously, but you're suggesting we change the vscode-nls and upgrade to l10n right now in all packages?

Reply:
We would have to update our vscode minimum version if we wanted to do this: https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/package.json#L12. vscode-l10n only supports 1.73.0 and above I think

Reply:
This PR suggests to add support for amazonqFeatureDev, not replace the library for all packages.

Reply:
Yea, otherwise, we need do it again once l10n is available for us.

Reply:
I think we looked at latest l10n version. It is already in our current vscode version. https://github.com/aws/aws-toolkit-vscode/blob/master/package-lock.json#L17394

v0.0.13

Reply:
This link sounds a 2nd level dependency. We're not using it directly.